### PR TITLE
[SCOTCH] Add builder

### PR DESCRIPTION
**Pull request by amontoison**

### CODE REVIEW

**Comment by amontoison** [comment]
@giordano It seems that we need a native build of  `dummysizes`. Is it possible to do a native build like `mkdefs`  (#351) ?

**Comment by giordano** [comment]
The host compilers are `${CC_BUILD}`, `${CXX_BUILD}`, etc..., but how to use them really depends on the build system.

**Comment by amontoison** [comment]
@giordano The header `sys/resource.h` is missing on `mingw` platforms. Do you have a solution or we remove Windows platforms?

**Comment by giordano** [comment]
That's a POSIX-only header file, not available for Windows (not in mingw nor MSVC): https://www.gnu.org/software/gnulib/manual/html_node/sys_002fresource_002eh.html. Where is this used? Perhaps there is a wrong guard? Or the package just can't possibly work on Windows?

**Comment by amontoison** [comment]
I fixed the issue with `sys/resource.h` but I have new errors now...
It's possible to compile it on Windows because `SCOTCH` have options to fix some issues on this platform, like `-DCOMMON_WINDOWS` but it remains difficult.

**Comment by giordano** [comment]
Then some guards must be wrong if it was trying to include a Unix-only header file

**Comment by amontoison** [comment]
I removed Windows platforms.

**Comment by giordano** [comment]
I could get past that error by adding `library_error$(OBJ)` to `LIBSCOTCHDEPS` in `src/libscotch/Makefile`, but then got more errors related to undefined references to `waitpid` which again a Posix function.  I wonder if they actually ever built for Windows, there are too many errors.

File: S/SCOTCH/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+cd src
+make
+make prefix=$prefix install
+install_license ../LICENSE_en.txt
+exit
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = filter!(p -> !Sys.iswindows(p), supported_platforms())
```
Comment (giordano) [inline S/SCOTCH/build_tarballs.jl:34]: For the record, you can also write
```julia
platforms = filter!(!Sys.iswindows, supported_platforms())
```
:wink:

**Comment by amontoison** [comment]
Thanks @giordano ! I don't think that they ever built for Windows....

A quick question, if I want to build `qr_mumps` with `SCOTCH` on all platforms except Windows, can I add `Dependency(PackageSpec(name="SCOTCH_jll", ...)` directly or I need to specify that this dependency is not needed (and not available in that case) for Windows platforms ?

**Comment by giordano** [comment]
Just add the dependency, nothing special needed

**Pull request merged**